### PR TITLE
redesign post-list

### LINKED ISSUE #776: List Page: Updates to cards for posts
See https://invis.io/CTLC400X932#/301261181_Home_Nav_Drawer_Open

Zeplin: https://app.zeplin.io/project/5b181d3bf71408ae4d406b94/screen/5b181e38cc5680ae11702acf

### acceptance criteria
- [x] remove user profile image
- [x] restyle text
- ~~restyle vote button / vote count~~
- [x] restyle comment button / comment count
- [ ] any click on the card opens the detail view (except clicking on the "new window" icon)
- [x] overflow menu for remaining buttons

### related issues (out of scope)
- user "headline" (e.g. "UCLA Post-doc Physics")
- Display thumbnail image for link-type posts that have thumbnails
- restyle vote button / vote count

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

closes #776 

#### What's this PR do?

Implements a redesign of the post list, which we use on the channel page and homepage and on the channel moderation page.

#### How should this be manually tested?

It should look alright and implement the things the design is supposed to have!

Also, check that all the functionality we currently offer on those pages still works alright.

### CODE REVIEW

**Comment by noisecapella** [comment]
I know this is one of those unsolved problems in computing but the select doesn't look right on firefox: 
screenshot from 2018-06-25 16-29-29

**Comment by noisecapella** [comment]
In the mockup is this supposed to be the icon for a text post? cc @roberthouse54 

screenshot from 2018-06-25 16-36-49

**Comment by noisecapella** [comment]
Functionality looks great! I'll look over code and tests tomorrow

File: static/js/components/UserMenu_test.js [hunk 1]
```diff
@@ -62,12 +59,14 @@ describe("UserMenu", () => {
...
-        assert.isOk(wrapper.find(DropdownWithClickOutside).exists())
+        assert.isOk(wrapper.find(DropdownMenu).exists())
       } else {
-        assert.isNotOk(wrapper.find(DropdownWithClickOutside).exists())
+        assert.isNotOk(wrapper.find(DropdownMenu).exists())
       }
     })
   })
+
+  //
```
Comment (noisecapella) [inline static/js/components/UserMenu_test.js]: Stray comment

File: static/js/components/UserMenu_test.js [hunk 2]
```diff
@@ -93,6 +93,8 @@ describe("UserMenu", () => {
     const wrapper = renderUserMenu()
     assert.isNotOk(wrapper.find(".profile-incomplete").exists())
   })
+
+  //
```
Comment (noisecapella) [inline static/js/components/UserMenu_test.js]: Stray comment

File: static/js/containers/ChannelPage.js [hunk 3]
```diff
@@ -144,17 +143,14 @@ class ChannelPage extends React.Component<*, void> {
...
-              <div className="post-list-title">
-                <div>{channel.title}</div>
-                <PostSortPicker
-                  updateSortParam={updatePostSortParam(this.props)}
-                  value={qs.parse(search).sort || POSTS_SORT_HOT}
-                />
-              </div>
-            }
-          >
+          <div>
```
Comment (noisecapella) [inline static/js/containers/ChannelPage.js]: Would this be a good place to use `React.Fragment` instead?

**Comment by noisecapella** [comment]
One other thing, the dropdown is empty for anonymous users because they can't do any of the actions. Should we remove the button in that case? Otherwise it looks a little strange:
screenshot from 2018-06-26 14-48-05

File: static/js/components/UserMenu_test.js (hunk 1 above)
Comment (aliceriot) [reply static/js/components/UserMenu_test.js]: yeah, I put these in intentionally to force prettier to leave a blank line between tests declare with an array, just cause it annoys me, haha

File: static/js/containers/ChannelPage.js (hunk 3 above)
Comment (aliceriot) [reply static/js/containers/ChannelPage.js]: yeah, good idea! The endless battle against extra `<div>` tags

**Comment by aliceriot** [comment]
ok, so I removed the menu button for anonymous users and switched over to react.fragment in that spot. I can't find where Lucas got the icon for the 'text' post from, so I think we can just leave it off for now and when he lets me know I can add it. As for firefox :woman_shrugging: I googled a little but didn't find anything super helpful

**Comment by aliceriot** [comment]
@noisecapella I just added a tiny change, if you want to take a quick look

**Comment by noisecapella** [comment]
Finally got my laptop set up to test this. Looks great! :+1:

**Pull request merged**